sentry-uitest/sentry-uitest-android-benchmark/src/androidTest/java/io/sentry/uitest/android/benchmark/SentryBenchmarkTest.kt

```diff
@@ -0,0 +1,137 @@
+package io.sentry.uitest.android.benchmark
+
+import android.content.Context
+import android.view.Choreographer
+import androidx.lifecycle.Lifecycle
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.core.app.launchActivity
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.IdlingRegistry
+import androidx.test.espresso.action.ViewActions.swipeUp
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.runner.AndroidJUnitRunner
+import io.sentry.ITransaction
+import io.sentry.Sentry
+import io.sentry.SentryOptions
+import io.sentry.android.core.SentryAndroid
+import io.sentry.uitest.android.benchmark.util.BenchmarkOperation
+import org.junit.Test
+import org.junit.runner.RunWith
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.assertTrue
+
+@RunWith(AndroidJUnit4::class)
+class SentryBenchmarkTest {
+
+    private lateinit var runner: AndroidJUnitRunner
+    private lateinit var context: Context
+    private lateinit var choreographer: Choreographer
+
+    @BeforeTest
+    fun setUp() {
+        runner = InstrumentationRegistry.getInstrumentation() as AndroidJUnitRunner
+        context = ApplicationProvider.getApplicationContext()
+        context.cacheDir.deleteRecursively()
+        IdlingRegistry.getInstance().register(BenchmarkActivity.scrollingIdlingResource)
+        // Must run on the main thread to get the main thread choreographer.
+        runner.runOnMainSync {
+            choreographer = Choreographer.getInstance()
+        }
+    }
+
+    @AfterTest
+    fun cleanup() {
+        IdlingRegistry.getInstance().unregister(BenchmarkActivity.scrollingIdlingResource)
+    }
+
+    @Test
+    fun benchmarkSameOperation() {
```

Comment:
Let's name the methods using the `like this`, so we keep the pattern with the other tests

Reply:
do you mean naming the method e.g. ``` `benchmark same operation` ```?
It's not possible here due to `Identifier not allowed in Android projects`
Or do you mean something else?

Reply:
Yes, the naming convention.
I just tested it and it worked on sentry-android-core.

Reply:
sentry-android-core is defined in gradle witht the plugin `com.android.library`.
When using the plugin `com.android.application` you cannot do that.
The `com.android.library` doesn't have the `isDebuggable` flag, and it seems it ignores what's in the Android manifest.

I found a workaround, though. I can change it to `com.android.library`, but i will have to add `(this as com.android.build.gradle.internal.dsl.BuildType).isDebuggable = false` in the `buildTypes` to make it not debuggable (just tested it and it's working fine).

Reply:
i tried setting the project as a `com.android.library`, as `sentry-android-core`, to check if it changes anything, but the method identifiers cannot contain spaces. 
It seems this behaviour involves classes in the `androidTest` folder

Reply:
It could be, thanks.
https://github.com/android/performance-samples/blob/main/MicrobenchmarkSample/microbenchmark/src/androidTest/java/com/example/benchmark/ViewInflateBenchmark.kt